spopt/locate/flow.py

```diff
@@ -0,0 +1,2579 @@
+                    if i + 1 in y and pulp.value(y[i + 1]) > 0.5:
+                        self.flow_coverage[od_pair] = {
+                            "flow_volume": self.flows[od_pair],
+                            "covered_proportion": 1.0,
+                            "covered_volume": self.flows[od_pair],
+                        }
+                    else:
+                        self.flow_coverage[od_pair] = {
+                            "flow_volume": self.flows[od_pair],
+                            "covered_proportion": 0.0,
+                            "covered_volume": 0.0,
+                        }
+
+                if self.threshold > 0:
+                    self.covered_nodes = [j for j in c if pulp.value(c[j]) > 0.5]
+                else:
+                    self.covered_nodes = []
+
+            coverage_percentage = covered_flow / total_flow if total_flow > 0 else 0
+
+            self.solver_stats = {
+                "solver_name": str(solver_instance),
+                "num_variables": len(model.variables()),
+                "num_constraints": len(
+                    [constraint for constraint in model.constraints.values()]
+                ),
+                "objective_sense": "maximize",
+            }
+
+            result = {
+                "status": self.status,
+                "objective_value": self.objective_value,
+                "selected_facilities": self.selected_facilities,
+                "flow_coverage": self.flow_coverage,
+                "total_flow": total_flow,
+                "covered_flow": covered_flow,
+                "coverage_percentage": coverage_percentage,
+                "solution_time": self.solution_time,
+            }
+
+            if self.capacity is None and self.threshold > 0:
+                result["covered_nodes"] = self.covered_nodes
+
+            if self.capacity is not None and objective == "vmt":
+                total_vmt = sum(
+                    self.flows[od_pair] * path_distances[od_pair]
+                    for od_pair in self.flows
+                )
+                covered_vmt = sum(
+                    coverage.get("covered_vmt", 0)
+                    for coverage in self.flow_coverage.values()
+                )
+                vmt_coverage_percentage = (
+                    covered_vmt / total_vmt if total_vmt > 0 else 0
+                )
+
+                result.update(
+                    {
+                        "total_vmt": total_vmt,
+                        "covered_vmt": covered_vmt,
+                        "vmt_coverage_percentage": vmt_coverage_percentage,
+                        "objective_type": objective,
+                    }
+                )
+
+            return result
+        else:
+            print(f"No optimal solution found. Status: {self.status}")
+            return {"status": self.status}
+
+    def _evaluate_capacitated_solution(self, facilities: Dict[Any, int]) -> Dict:
```

Comment:
@fengzixin0617 I did not understand this method. We are already defining the capacity constraints within `solve_pulp` method but here are we defining it again?